a4_APIweatherApp.py

```diff
@@ -0,0 +1,90 @@
+'''
+This program is inteded to use the Open Meteo API
+and output an API and Mock plot of temperatures in the next 24 hours
+'''
+
+from abc import ABC, abstractmethod
+import requests
+import matplotlib.pyplot as plt
+
+class Service(ABC):
+    @abstractmethod
+    def getting(self):
+        pass
+
+class mockService(Service):
+    def getting(self):
+        #create a 
+        dates = ["2024-08-02T02:00",
+      "2024-08-02T03:00",
+      "2024-08-02T04:00",
+      "2024-08-02T05:00",
+      "2024-08-02T06:00",
+      "2024-08-02T07:00",
+      "2024-08-02T08:00",
+      "2024-08-02T09:00",
+      "2024-08-02T10:00",
+      "2024-08-02T11:00",
+      "2024-08-02T12:00",
+      "2024-08-02T13:00",
+      "2024-08-02T14:00",
+      "2024-08-02T15:00",
+      "2024-08-02T16:00",
+      "2024-08-02T17:00",
+      "2024-08-02T18:00",
+      "2024-08-02T19:00",
+      "2024-08-02T20:00",
+      "2024-08-02T21:00",
+      "2024-08-02T22:00",
+      "2024-08-02T23:00",
+      "2024-08-03T00:00",
+      "2024-08-03T01:00"]
+        temps = [19.4, 18.8, 18.2, 17.8, 17.7, 17.7, 17.9, 18, 18.1, 18.6, 18.8, 18.8, 18.9, 19.2, 19.4, 19.6, 19.4, 19.5, 19.4, 19, 18.5, 18, 17.6, 17.3]
+        plt.scatter(dates, temps)
```

Comment:
The idea behind this function is correct, but I don't think we need to add these lines about plotting here. I would keep this as a separate function/class, and have the `getting` function just return the values directly. The plotting functionality makes more sense to have as something that happend in the `Handler` class, since the responsibility of the `Service` objects should be to just get data.

When you are writing a function, think about what is it that a given function is responsible for. Can you easily write that in one or two sentences? Its helpful to keep that description as a docstring in your function as well, so if you end up adding more, its clear what should move out, and what should be kept in.